### PR TITLE
docs: add GitHub app integration documentation

### LINKED ISSUE #41: Add GitHub app integration documentation
## Overview

Add a new integration guide for the existing [Phylum GitHub App](https://github.com/marketplace/phylum-io).

## Additional context

Provide more detail around the new content:

* The content will be like the "Current integrations" currently linked from the [integrations overview page](https://docs.phylum.io/docs/integrations_overview)
  * The current link there should be updated to point to this new guide
* The page name could be `GitHub App Integration`
* The address/slug could be
  * https://docs.phylum.io/docs/github_app now for Readme.com
  * https://docs.phylum.io/docs/integrations/github_app going forward for Docusaurus
* Include screenshots to walk through the complete process
* Include any/all pre-requisite steps
* Where there are options (e.g., linking a federated identity) show the most common path in detail

### PR DESCRIPTION
**Pull request by maxrake**
This change adds documentation for the GitHub app integration.

Actions taken include:

* Add `integrations/github_app.md`
* Add screenshots to the `assets` directory
  * Refer to the screenshots by the location they _will_ exist on `main`
* Update the `integrations_overview.md` to point to new page
* Add knowledge base pages
  * Account federation info
  * How to transfer group ownership
* Format `knowledge_base/faq.md`

Closes #41

### CODE REVIEW

File: docs/integrations/github_app.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+That will provide results and details for individual package issues. Each issue can be reviewed and suppressed if it is found to be a false positive, irrelevant, or otherwise acceptable to proceed:
+
+![Phylum app issue suppression](https://raw.githubusercontent.com/phylum-dev/documentation/main/assets/issue_suppression.png)
+
+Remaining issues may require lockfile modification to find alternative dependencies or otherwise remove the offending package.
+
+## FAQ
+
+1. Q: I activated monitoring, but it didn't run a scan. How do I get analysis results?
+   * A: Activating monitoring tells the app to start watching for commits in a PR that modify supported lockfiles. To see scan results, start committing code!
```
Comment (furi0us333) [inline docs/integrations/github_app.md:114]:
```suggestion
   * A: Check to ensure the repository contains a supported lockfile.
```

The application has been updated to automatically scan the default branch when monitoring is activated.

File: docs/integrations/integrations_overview.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ The current Continuous Integration (CI) platforms/environments supported are:
 
 | Platform/Environment | Information Link |
 | -------------------- | ---------------- |
-| GitHub App | [GitHub Marketplace][github_app_docs] |
+| GitHub App | [Documentation][github_app_docs] |
```
Comment (furi0us333) [inline docs/integrations/integrations_overview.md:17]: I actually think we should have both the doc and marketplace links. You can separate with a semicolon or whatever looks best to differentiate.

File: docs/integrations/github_app.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+## Usage
+
+Once installed, the GitHub app can be managed through the settings menu available from the [Phylum UI](https://app.phylum.io). Click on your user icon and select the `GitHub App Settings` option from the dropdown menu:
+
+![Phylum GitHub app settings dropdown](https://raw.githubusercontent.com/phylum-dev/documentation/main/assets/gh_app_settings_menu.png)
+
+### Monitoring
+
+Monitoring can be activated or paused by selecting the toggle for a given repository. When first activated, a Phylum Project will be created.
```
Comment (furi0us333) [inline docs/integrations/github_app.md:43]:
```suggestion
Monitoring can be activated or paused by selecting the toggle for a given repository. When first activated, a Phylum Project will be created and any supported lockfiles in the default branch will be analyzed.
```

File: docs/integrations/github_app.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+5. Choose **All** (default) or **Select** repositories and click to install. This setting controls which repositories _can be_ monitored by Phylum (i.e., visibility). Configuring repositories to be monitored [is a different step](#monitoring).
+
+   ![GitHub App install confirmation screen](https://raw.githubusercontent.com/phylum-dev/documentation/main/assets/gh_app_install_2.png)
+
+6. After installation, you will be redirected to the Phylum application to configure monitoring.
+7. If you are a new user or are not logged in, select the GitHub button at the login screen.
+   * This will automatically provision an account for new users.
+8. Congratulations - the Phylum GitHub App is installed and ready to be configured to monitor your software supply chain!
+
+## Usage
```
Comment (furi0us333) [inline docs/integrations/github_app.md:35]: Now that the on-demand analysis button has landed, we should probably have a section explaining that as well.

File: docs/integrations/github_app.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+![Phylum groups management](https://raw.githubusercontent.com/phylum-dev/documentation/main/assets/phylum_groups_management.png)
+
+The default group that is created is owned by the account that installed the GitHub App. If you would like a different account to manage the GitHub App and group you may [transfer ownership](https://docs.phylum.io/docs/transfer_group_ownership) of the group.
+
+### Policy
+
+The Phylum GitHub app uses the [established project policy](https://docs.phylum.io/docs/policy) for making overall success/failure risk decisions. No configuration is required for setting the policy since a default policy is used for all projects. However, [Phylum PRO users](https://www.phylum.io/pricing) may specify custom policies for their projects to exercise fine-grained control over the risk decision logic.
+
+### Remediation
```
Comment (furi0us333) [inline docs/integrations/github_app.md:99]: Should we mention re-triggering the analysis once the remediations have been made?

**Review by furi0us333 — CHANGES_REQUESTED** [review]

File: docs/integrations/github_app.md (hunk 5 above)
Comment (maxrake) [reply docs/integrations/github_app.md:99]: Yes...no problem...change coming...

File: docs/integrations/github_app.md (hunk 4 above)
Comment (maxrake) [reply docs/integrations/github_app.md:35]: Yes...no problem...change coming...

**Review by furi0us333 — APPROVED** [review]
LGTM

**Pull request merged**